sdk/python/kfp/components/pipeline_channel.py

```diff
@@ -253,35 +251,48 @@ def __init__(
 
 def create_pipeline_channel(
     name: str,
-    channel_type: Union[str, Dict],
+    spec: Union[structures.InputSpec, structures.OutputSpec],
     task_name: Optional[str] = None,
-    value: Optional[type_utils.PARAMETER_TYPES] = None,
 ) -> PipelineChannel:
     """Creates a PipelineChannel object.
 
     Args:
         name: The name of the channel.
-        channel_type: The type of the channel, which decides whether it is an
+        spec: The InputSpec or OutputSpec corresponding to this channel. This
+            determines whether the constructed channel is a
             PipelineParameterChannel or PipelineArtifactChannel
         task_name: Optional; the task that produced the channel.
-        value: Optional; the realized value for a channel.
 
     Returns:
         A PipelineParameterChannel or PipelineArtifactChannel object.
     """
-    if type_utils.is_parameter_type(channel_type):
-        return PipelineParameterChannel(
-            name=name,
-            channel_type=channel_type,
-            task_name=task_name,
-            value=value,
-        )
+    if isinstance(spec, structures.InputSpec):
+        if type_utils.is_parameter_type(spec.type):
+            return PipelineParameterChannel(
+                name=name,
+                channel_type=spec.type,
+                task_name=task_name,
+                default_value=spec.default,
+            )
+        else:
+            return PipelineArtifactChannel(
+                name=name,
+                channel_type=spec.type,
+                task_name=task_name,
+            )
     else:
-        return PipelineArtifactChannel(
-            name=name,
-            channel_type=channel_type,
-            task_name=task_name,
-        )
+        if type_utils.is_parameter_type(spec.type):
+            return PipelineParameterChannel(
+                name=name,
+                channel_type=spec.type,
+                task_name=task_name,
+            )
+        else:
+            return PipelineArtifactChannel(
+                name=name,
+                channel_type=spec.type,
+                task_name=task_name,
+            )
```

Comment:
The way `PipelineArtifactChannel` is created here isn't perfectly DRY, but I'm opting to leave this split since they will be constructed differently when we support optional input artifacts.

Reply:
I would suggest we combine the change with the actual optional artifact handling change together. Otherwise the current change by itself looks going backward of the purpose of refactoring.

Reply:
Agreed and will do.